### PR TITLE
fix(curriculum): explain media queries

### LINKED ISSUE #46886: When/when are RWD media queries first taught?
### Describe the Issue

Edit: I'm talking about a standard media query used in responsive design.

Is [step 27](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-responsive-web-design-by-building-a-piano/step-27) in "Learn Responsive Web Design by Building a Piano" really the first mention? And if so how does that step qualify as teaching media queries? Nothing is explained at all. If this challenge step is meant to be the first introduction to RWD media queries then it should be updated with an explanation and syntax example.

[Step 114](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-css-variables-by-building-a-city-skyline/step-114) in "Learn CSS Variables by Building a City Skyline" at least explains it briefly and has a syntax example. 

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-responsive-web-design-by-building-a-piano/step-27

### Your code

```
@media (max-width: 960px) {
  .card {
    flex-direction: column;
  }
}
```

### Expected behavior

Explain what a media query is, what it looks like, and how it works.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lasjorg**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46886

Add media query explanation and code example.

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/learn-responsive-web-design-by-building-a-piano/612ebcba99bfa46a15370b11.md [hunk 1]
```diff
@@ -7,7 +7,19 @@ dashedName: step-27
 
 # --description--
 
-Now you need to make it responsive. Add a `@media` query with a `max-width` of `768px`.
+The `@media` at-rule, also known as a media query, is used to conditionally apply CSS. Media queries are commonly used to apply CSS based on the viewport width using the `max-width` and `min-width` properties.
+
+In the below example the padding is applied to the `.card` class when the viewport is `960px` wide and below.
+
+```css
+@media (max-width: 960px) {
```
Comment (jeremylt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-responsive-web-design-by-building-a-piano/612ebcba99bfa46a15370b11.md:15]: Two questions 

1) Do we want to demo min-width and ask for max-width?

2) Do you think users will randomly insert the CSS rule for the card class? Do we need to add a test for that?

Comment (lasjorg) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-responsive-web-design-by-building-a-piano/612ebcba99bfa46a15370b11.md:15]: 1. I think it is better not to confuse people and keep the example using `max-width`
2. It doesn't really matter what selector they put inside. It works with or without a selector inside the media query. The test doesn't check for that and it isn't part of the requirement. I'm fine with them being able to just copy-pasting the example and change the px value in the condition to pass the step.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-responsive-web-design-by-building-a-piano/612ebcba99bfa46a15370b11.md [hunk 2]
```diff
@@ -7,7 +7,19 @@ dashedName: step-27
...
+  .card {
+    padding: 2rem;
+  }
+}
+```
+
+Add a `@media` query with a `max-width` of `768px`.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-responsive-web-design-by-building-a-piano/612ebcba99bfa46a15370b11.md:22]:
```suggestion
Add a media query that will be applied when the viewport is `768px` wide and below.
```

Not required, but I wonder if we should change this to match the verbiage of the example so the camper has to think a bit about what they are actually writing.

Comment (lasjorg) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-responsive-web-design-by-building-a-piano/612ebcba99bfa46a15370b11.md:22]: I wouldn't be opposed to that. The hint has most of the syntax and the example should help as well.

**Review by naomi-lgbt — APPROVED** [review]
I'm happy with this.

Nice new avatar, btw.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good

**Comment by lasjorg** [comment]
> Nice new avatar, btw.

Thanks, I finally got tired of looking at the default one.

**Review by Sembauke — APPROVED** [review]

**Pull request merged**